### DISCUSSION TITLE
Why PointerEventArgs.Properties is not public?

### DISCUSSION THREAD

**Discussion by artemiusgreat** [AvaloniaUI/Avalonia#7056] [Q&A] [answered]
To implement Drag'n'Drop, I would need to intercept mouse move event and at the same time I need to know if left mouse button is pressed to know when to start dragging. 

I create mouse move event handler but the only property that can tell if mouse button is pressed is not accessible. 

```
PointerMoved += (object sender, PointerEventArgs e) => 
{
  // e.Properties.isLeftMouseButtonPressed - here Properties is protected and not accessible
};
```
Alternatively, it's possible to use [other events handlers](https://stackoverflow.com/a/66388635/437393
) to track mouse button pressing. While it's possible to combine mouse click events with mouse move events to create Drag'n'Drop, I was wondering, **why information about mouse buttons is intentionally not available inside mouse move handler**?

**Comment by kekekeks** [comment, marked as answer]
You are supposed to use `e.GetCurrentPoint(this).Properties`

**Reply by kekekeks** [reply]
Rationale: we are planning to combine multiple pointer points into one event later like UWP does.
